build.cmd

```diff
@@ -6,10 +6,20 @@ setlocal
 ::       means that that rebuilding cannot successfully delete the task
 ::       assembly.
 
+:ReadArguments
+:: Read in the args to determine whether to run the native build, managed build, or both (default)
+set "__args=%*"
+set "__buildSpec=both"
+if /i [%1] == [both]    	(set __buildSpec=both&&set "__args=%__args:~4%"&&shift&&goto Tools)
+if /i [%1] == [native]    	(set __buildSpec=native&&set "__args=%__args:~6%"&&shift&&goto Tools)
+if /i [%1] == [managed]    	(set __buildSpec=managed&&set "__args=%__args:~7%"&&shift&&goto Tools)
```

Comment:
interesting I didn't realize that was the case, but I just did a quick test and your correct. I guess in this case shift or the :ReadArguments label isn't even needed because you are only parsing one argument. However I'm OK with keeping them the it in place.